src/topology/homotopy/product.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2021 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import topology.homotopy.basic
+import topology.constructions
+
+/-!
+# Product of homotopies
+
+In this file, we introduce definitions for the product of
+homotopies. We show that the product of relative homotopies
+are still relative homotopies.
+
+## Definitions
+- `product_homotopy f g S homotopies`: Let f and g be a family of functions
+  indexed on I, such that for each i ∈ I, fᵢ and gᵢ are maps from A to Xᵢ.
+  Let `homotopies` be a family of homotopies from fᵢ to gᵢ for each i.
+  Then `product_homotopy f g S homotopies` is the canonical homotopy
+  from ∏ f to ∏ g, where ∏ f is the product map from A to Πi, Xᵢ.
+  All homotopies are done relative to the set S ⊆ A.
+-/
+
+noncomputable theory
+
+namespace continuous_map
+section
+parameters {I : Type*} {X : I → Type*}
+           [∀i, topological_space (X i)]
+           {A : Type*}
+           [topological_space A]
+
+/-- Abbreviation for product of continuous maps, which is continuous -/
+def pi (f : Π i, C(A, X i)) : C(A, Π i, X i) :=
+{ to_fun := λ (a : A) (i : I), f i a,
+  continuous_to_fun := by continuity, }
+
+@[simp]
+lemma pi_eval (f : Π i, C(A, X i)) (a : A)  :
+  (pi f) a  = λ i : I, (f i) a := rfl
+
+namespace homotopy
+/-- The product homotopy of `homotopies` between functions `f`
```

Comment:
Could you please also add a version for binary products? I.e., when `H₁` and `H₂` are two homotopies `f₁ ~ g₁` and `f₂ ~ g₂`.

Reply:
I just duplicated the result instead of trying to plug in `I = fin 2` or something. Not sure what the best way to do it is.